src/form/Form.js

```diff
@@ -196,6 +215,49 @@
         this.btntd
             .attr("colspan", this._maxCols * 2)
         ;
+        
+        // Disable Submit unless there is data
+        if (!this.allowEmptyRequest()) {
+            this.inputs().forEach(function(input, idx) {
+                if (input._inputElement instanceof Array) {
+                    input._inputElement.forEach(function(e) {
+                        e.on("change.form", function(w) {
+                            setTimeout(function() {
+                                var hasValue = context.inputs().some(function(w) {
+                                    if (w.value() && w.value !== "false") {
+                                        return true;
+                                    }
+                                    return false;
+                                });
+                                if (hasValue) {
+                                    context._controls[0]._inputElement[0].attr("disabled", null);
+                                } else {
+                                    context._controls[0]._inputElement[0].attr("disabled", "disabled");
+                                }
+                            }, 100);
+                        });
+                    });
+                }
+            });
+        }
+    };
+
+    Form.prototype.render = function (callback) {
+        var retVal =  HTMLWidget.prototype.render.apply(this, arguments);
+        if (!this.allowEmptyRequest()) {
```

Comment:
use a settimeout here too ... that way we give the form enough time to run updates and all before checking